src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/CommandHandlers/SendEmailToEmployerCommandHandler.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NServiceBus;
+using SFA.DAS.CommitmentsV2.Messages.Commands;
+using SFA.DAS.EAS.Account.Api.Client;
+using SFA.DAS.Notifications.Messages.Commands;
+
+namespace SFA.DAS.CommitmentsV2.MessageHandlers.CommandHandlers
+{
+    public class SendEmailToEmployerCommandHandler : IHandleMessages<SendEmailToEmployerCommand>
+    {
+        private const string Owner = "Owner";
+        private const string Transactor = "Transactor";
+        private const string ReplyTo = "noreply@sfa.gov.uk";
+
+        private readonly ILogger<SendEmailToEmployerCommandHandler> _logger;
+        private readonly IAccountApiClient _accountApiClient;
+
+        public SendEmailToEmployerCommandHandler(IAccountApiClient accountApiClient, ILogger<SendEmailToEmployerCommandHandler> logger)
+        {
+            _logger = logger;
+            _accountApiClient = accountApiClient;
+        }
+
+        public async Task Handle(SendEmailToEmployerCommand message, IMessageHandlerContext context)
+        {
+            try
+            {
+                List<string> emails;
+
+                bool IsOwnerOrTransactor(string role)
+                {
+                    return role.Equals(Owner, StringComparison.InvariantCultureIgnoreCase) ||
+                           role.Equals(Transactor, StringComparison.InvariantCultureIgnoreCase);
+                }
+
+                if (string.IsNullOrWhiteSpace(message.EmailAddress))
+                {
+                    var users = await _accountApiClient.GetAccountUsers(message.AccountId);
+                    emails = users.Where(x =>
+                            x.CanReceiveNotifications && !string.IsNullOrWhiteSpace(x.Email) &&
+                            IsOwnerOrTransactor(x.Role))
+                        .Select(x => x.Email).ToList();
+                }
+                else
+                {
+                    emails = new List<string> {message.EmailAddress};
```

Comment:
I believe we still need to call the account api here in order to establish whether the user has opted out. If sending to a specific user and they have opted out, no-one gets the email.

But, for future-proofing, could we also add a property to the command that allows us to override the opt-out? This would be for really important emails we want to make sure they get, no matter what.